### PR TITLE
added tests and fixed GradPow, GradMatMul

### LINKED ISSUE #206: add tests for GradMatMul
## Description
Tests for GradMatMul is missing (both forward and backward).

## Type of Test
- [x] **Unit test** (e.g. checking a loop, method, or function is working as intended)

### LINKED ISSUE #210: add tests for GradPow
## Description
Tests for GradPow are missing (both forward and backward).

## Type of Test
- [x] **Unit test** (e.g. checking a loop, method, or function is working as intended)

### PR DESCRIPTION
**Pull request by aanurraj**
## Description
this PR closes #210  closes  #206 

## Affected Dependencies
None

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by gmuraru** [comment]
Only add an exception for the case we use a "Y" which is not an integer - such that we make sure that the gradient is propagated only to the "X" variable.

**Review by gmuraru — CHANGES_REQUESTED** [review]
See comment

**Review by LaRiffle — APPROVED** [review]
🙌

**Review by Syzygianinfern0 — COMMENTED** [review]
Do we have support for tensor with `no_grad=True`?

**Comment by gmuraru** [comment]
> Do we have support for tensor with `no_grad=True`?

I don't think we have

File: tests/sympc/tensor/grad/grad_functions_test.py [hunk 1]
```diff
@@ -459,3 +461,94 @@ def test_grad_flatten_backward(get_clients) -> None:
...
+@pytest.mark.parametrize("power", [2, 4, 5, 1.0])
+def test_grad_pow_forward(get_clients, power) -> None:
+    parties = get_clients(4)
+    x = torch.Tensor([[1, 2, 3], [4, 5, 6]])
+
+    x_mpc = x.share(parties=parties)
+
+    ctx = {}
+
+    if isinstance(power, float):
```
Comment (gmuraru) [inline tests/sympc/tensor/grad/grad_functions_test.py:475]: MAYBE: Create a separate test to "test" the exception path. Each test should have a single logic to be tested.

**Review by gmuraru — CHANGES_REQUESTED** [review]
Check comment

File: tests/sympc/tensor/grad/grad_functions_test.py [hunk 2]
```diff
@@ -459,3 +461,104 @@ def test_grad_flatten_backward(get_clients) -> None:
...
+@pytest.mark.parametrize("power", [1.0, torch.tensor([1, 3])])
+def test_grad_pow_forward_exception(get_clients, power) -> None:
+    parties = get_clients(4)
+    x = torch.Tensor([[1, 2, 3], [4, 5, 6]])
+
+    x_mpc = x.share(parties=parties)
+
+    ctx = {}
+
+    if not isinstance(power, int):
```
Comment (gmuraru) [inline tests/sympc/tensor/grad/grad_functions_test.py:495]: Remove this line because we know they are not int

**Review by Syzygianinfern0 — APPROVED** [review]
🚀

**Review by gmuraru — APPROVED** [review]
LGTM!
